jsp-questions.md

```diff
@@ -436,7 +436,33 @@ The jsp page, by default, always creates a session. Using a directive pagewith a
 ## Q. What is the difference between JSPWriter and PrintWriter?
 `PrintWriter` is the object responsible for recording the contents of the response to the request. `JspWriter` uses an object `PrintWriter` to buffer. When the buffer is full or flushed, it `JspWriter`uses the object `PrintWriter` to write the content in response.
 
-#### Q. How to disable caching on back button of the browser?
+## Q. How to disable caching on back button of the browser?
+for this, once the session is invalidated, in your respective jsp page add following code snippet 
+```jsp
+<%
+    response.setHeader("Cache-Control","no-cache, no-store, must-revalidate");
+    response.setHeader("Pragma","no-cache");
+    response.setHeader ("Expires", 0);
+    response.setDateHeader ("Expires", -1);
+    if(session.getAttribute("token")==null){
+        response.sendRedirect("login.jsp");
```

Comment:
I feel, this would be more appropriate

`response.sendRedirect(request.getContextPath() + "/login.jsp");`

Reply:
made changes, with commit [8bf0706](https://github.com/learning-zone/java-interview-questions/pull/9/commits/8bf07065f7c6376ae8a47dc19dc352ac5240114a) ,